Cargo.toml

```diff
@@ -282,6 +282,10 @@ cfg-if = { version = "1.0.0", default-features = false }
 tonic = { version = "0.5", optional = true, default-features = false, features = ["transport", "codegen", "prost", "tls"] }
 data-encoding = { version = "2.2", default-features = false, features = ["std"], optional = true }
 trust-dns-proto = { version = "0.20", features = ["dnssec"], optional = true }
+mac_address = { version = "1.1.2", default-features = false, optional = true }
+global_counter = { version = "0.2.2", default-features = false, optional = true }
```

Comment:
Instead of adding a new dependency to the project -- especially one that hasn't reached 1.0 yet -- let's prefer the use of `AtomicU32`. Or, to jump the gun a little, prefer the use of `u32` in a simple struct.

Reply:
is using `u32` in a struct thread-safe?

Reply:
Also just realised that I don't handle overflows, so will have to add this as well, but that can be done as a simple reset for `u32` or `AtomicU32`:
```rust
if counter > 10000 {
  counter = 0;
}
```

Reply:
This code above has a race actually, so wasn't thread-safe at all, but AtomicU32 actually wraps around on overflow, so I switched to it.

Reply:
switched to u32 as this code is single-threaded.